src/Education/Commands/RuleHelpUserControl.xaml.cs

```diff
@@ -21,24 +21,35 @@
 using System.Windows.Controls;
 using System.Windows.Navigation;
 using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration;
 
 namespace SonarLint.VisualStudio.Education.Commands
 {
     public sealed partial class RuleHelpUserControl : UserControl
     {
         private readonly IBrowserService browserService;
         private readonly IEducation education;
+        private readonly ILogger logger;
 
-        internal RuleHelpUserControl(IBrowserService browserService, IEducation education)
+        internal RuleHelpUserControl(IBrowserService browserService, IEducation education, ILogger logger)
         {
             this.browserService = browserService;
             this.education = education;
+            this.logger = logger;
 
             InitializeComponent();
         }
 
         public void HandleRequestNavigate(object sender, RequestNavigateEventArgs e)
         {
+            if (!e.Uri.IsAbsoluteUri)
+            {
+                logger.LogVerbose($"[RuleHelpUserControl] Failed to navigate to Uri as it is a relative path: {e.Uri}");
```

Comment:
I wanted to use the resource file but no matter what I did this class would not recognize Resources so I gave up, Do I need to do something special to be able to reference it?

Reply:
Interesting - I hadn't come across this problem before.

This is a `UserControl` so it already has a `Resources` property so just referring to `Resources` won't work. I expected using the full-qualified name `SonarLint.VisualStudio.Education.Resources` in the code to work, but it doesn't (I'm not entirely sure why - there's no reason for it not to work at the compiler level, so it might be a bug in VS). I'm guessing you tried both of those.

I found a couple of workarounds:
* change the namespace of the control so it isn't under `SonarLint.VisualStudio.Education`.
* change the name of the resource file to something other than `Resources`
* change the namespace of the `Resources` file e.g. by setting the "Custom Tool Namespace" property in the `Properties` window

* last, and simplest: add a `using` alias in the `RuleHelpUserControl` e.g. 
`using EducationResx = SonarLint.VisualStudio.Education.Resources;`

You can then reference the resource file as `EducationResx.Resources`. `Using` alias are a way of handling namespace conflicts.